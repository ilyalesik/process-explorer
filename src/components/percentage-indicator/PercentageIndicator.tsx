import * as React from "react";
import * as d3 from "d3";

const percentageToColor = value => d3.interpolateSinebow(0.2 + value);

const toFixed2 = num => Math.round(num * 100) / 100;

export const PercentageIndicator = (props: { value: number; className?: string }) => {
	const { value, className } = props;
	return (
		<p style={{ color: React.useMemo(() => percentageToColor(value), [value]) }} className={className}>
			{toFixed2(value)}%
		</p>
	);
};
