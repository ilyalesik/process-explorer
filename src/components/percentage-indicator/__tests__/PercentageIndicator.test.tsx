import * as React from "react";
import * as renderer from "react-test-renderer";
import { PercentageIndicator } from "../PercentageIndicator";

describe("<PercentageIndicator />", () => {
	it("PercentageIndicator renders correctly with value = 0", () => {
		const tree = renderer.create(<PercentageIndicator value={0} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("PercentageIndicator renders correctly with value = 0.1", () => {
		const tree = renderer.create(<PercentageIndicator value={0.1} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("PercentageIndicator renders correctly with value = 0.66666", () => {
		const tree = renderer.create(<PercentageIndicator value={0.66666} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("PercentageIndicator renders correctly with value = 100.1", () => {
		const tree = renderer.create(<PercentageIndicator value={100.1} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
