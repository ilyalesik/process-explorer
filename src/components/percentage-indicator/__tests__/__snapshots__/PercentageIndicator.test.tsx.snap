// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PercentageIndicator /> PercentageIndicator renders correctly with value = 0 1`] = `
<p
  style={
    Object {
      "color": "rgb(167, 213, 3)",
    }
  }
>
  0
  %
</p>
`;

exports[`<PercentageIndicator /> PercentageIndicator renders correctly with value = 0.1 1`] = `
<p
  style={
    Object {
      "color": "rgb(88, 252, 42)",
    }
  }
>
  0.1
  %
</p>
`;

exports[`<PercentageIndicator /> PercentageIndicator renders correctly with value = 0.66666 1`] = `
<p
  style={
    Object {
      "color": "rgb(213, 3, 167)",
    }
  }
>
  0.67
  %
</p>
`;

exports[`<PercentageIndicator /> PercentageIndicator renders correctly with value = 100.1 1`] = `
<p
  style={
    Object {
      "color": "rgb(88, 252, 42)",
    }
  }
>
  100.1
  %
</p>
`;
